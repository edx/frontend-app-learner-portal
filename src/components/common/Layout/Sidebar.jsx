import React from 'react';

const Sidebar = props => (
  <div className="col offset-lg-1">
    {props.children}
  </div>
);

export default Sidebar;
