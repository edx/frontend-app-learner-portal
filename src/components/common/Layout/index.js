// eslint-disable-next-line import/prefer-default-export
export { default as Layout } from './Layout';
export { default as MainContent } from './MainContent';
export { default as Sidebar } from './Sidebar';
export { default as SidebarBlock } from './SidebarBlock';
