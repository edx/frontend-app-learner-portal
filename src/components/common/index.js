export { default as Layout, LayoutConsumer } from './Layout/Layout';
export { default as withAuthentication } from './withAuthentication';
