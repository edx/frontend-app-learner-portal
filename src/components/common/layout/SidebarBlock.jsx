import React from 'react';
import PropTypes from 'prop-types';

const SidebarBlock = props => (
  <div className={props.className}>
    {props.title && <h2 className="mb-2">{props.title}</h2>}
    {props.children}
  </div>
);

SidebarBlock.propTypes = {
  title: PropTypes.string,
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
};

SidebarBlock.defaultProps = {
  title: null,
  className: undefined,
};

export default SidebarBlock;
