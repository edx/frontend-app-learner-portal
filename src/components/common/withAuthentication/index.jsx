import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { fetchUserAccount as _fetchUserAccount, UserAccountApiService } from '@edx/frontend-auth';
import {
  configureAnalytics,
  identifyAuthenticatedUser,
  identifyAnonymousUser,
  initializeSegment,
  sendPageEvent,
} from '@edx/frontend-analytics';
import { configureLoggingService, NewRelicLoggingService } from '@edx/frontend-logging';
import { StaticQuery, graphql } from 'gatsby';

import apiClient from '../../../apiClient';

const withAuthentication = (WrappedComponent) => {
  const ComponentClass = class extends Component {
    static propTypes = {
      location: PropTypes.shape({
        pathname: PropTypes.string.isRequired,
      }).isRequired,
      fetchUserAccount: PropTypes.func.isRequired,
      providerSlug: PropTypes.string.isRequired,
      username: PropTypes.string,
    };

    static defaultProps = {
      username: null,
    };

    state = {
      isLoading: true,
    };

    componentDidMount() {
      const {
        username, location, fetchUserAccount, providerSlug,
      } = this.props;
      apiClient.loginUrl = `${process.env.LMS_BASE_URL}/auth/idp_redirect/${providerSlug}`;
      apiClient.ensurePublicOrAuthenticationAndCookies(location.pathname, async (accessToken) => {
        configureLoggingService(NewRelicLoggingService);
        initializeSegment(process.env.SEGMENT_KEY);
        configureAnalytics({
          loggingService: NewRelicLoggingService,
          authApiClient: apiClient,
          analyticsApiBaseUrl: process.env.LMS_BASE_URL,
        });

        const userAccountApiService = new UserAccountApiService(
          apiClient,
          process.env.LMS_BASE_URL,
        );
        await fetchUserAccount(userAccountApiService, username);
        if (accessToken) {
          identifyAuthenticatedUser(accessToken.user_id);
        } else {
          identifyAnonymousUser();
        }
        sendPageEvent();
        this.setState({ isLoading: false });
      });
    }

    render() {
      const { isLoading } = this.state;
      if (isLoading) {
        return null;
      }
      return <WrappedComponent {...this.props} />;
    }
  };

  return connect(
    state => ({ username: state.authentication.username }),
    { fetchUserAccount: _fetchUserAccount },
  )(ComponentClass);
};

withAuthentication.propTypes = {
  location: PropTypes.string.isRequired,
  providerSlug: PropTypes.string.isRequired,
};

const withSaml = WrappedComponent => (
  props => (
    <StaticQuery
      query={graphql`
        query {
          site {
            siteMetadata {
              providerSlug
            }
          }
        }
      `}
      render={data => (
        <WrappedComponent providerSlug={data.site.siteMetadata.providerSlug} {...props} />
      )}
    />
  )
);

export default WrappedComponent => withSaml(withAuthentication(WrappedComponent));
