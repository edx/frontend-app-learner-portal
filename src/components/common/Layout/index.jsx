import React from 'react';
import { Helmet } from 'react-helmet';
import PropTypes from 'prop-types';
import SiteHeader from '@edx/frontend-component-site-header';
import SiteFooter from '@edx/frontend-component-footer';
import { connect } from 'react-redux';

import EdXLogo from '../../../images/edx-logo.svg';

import './styles/Layout.scss';

const Layout = props => (
  <>
    <Helmet titleTemplate="%s - edX" defaultTitle="edX" />
    <SiteHeader
      logo={EdXLogo}
      logoDestination={props.siteUrl}
      logoAltText={props.siteName}
      loggedIn={!!props.username}
      username={props.username}
      avatar={props.avatar}
      userMenu={[
        {
          type: 'item',
          href: process.env.LMS_BASE_URL,
          content: 'Dashboard',
        },
        {
          type: 'item',
          href: '/',
          content: 'My Masters Degree',
        },
        {
          type: 'item',
          href: `${process.env.LMS_BASE_URL}/u/${props.username}`,
          content: 'Profile',
        },
        {
          type: 'item',
          href: `${process.env.LMS_BASE_URL}/account/settings`,
          content: 'Account Settings',
        },
        {
          type: 'item',
          href: process.env.LOGOUT_URL,
          content: 'Sign out',
        },
      ]}
      loggedOutItems={[
        { type: 'item', href: '#', content: 'Login' },
        { type: 'item', href: '#', content: 'Sign Up' },
      ]}
      skipNavId="content"
    />
    <>{props.children}</>
    <SiteFooter
      siteName={props.siteName}
      siteLogo={EdXLogo}
      marketingSiteBaseUrl="https://www.example.com"
      supportUrl="https://www.example.com/support"
      contactUrl="https://www.example.com/contact"
      openSourceUrl="https://www.example.com/open"
      termsOfServiceUrl="https://www.example.com/terms-of-service"
      privacyPolicyUrl="https://www.example.com/privacy-policy"
      facebookUrl="https://www.facebook.com"
      twitterUrl="https://www.twitter.com"
      youTubeUrl="https://www.youtube.com"
      linkedInUrl="https://www.linkedin.com"
      googlePlusUrl="https://plus.google.com"
      redditUrl="https://reddit.com"
      appleAppStoreUrl="https://store.apple.com"
      googlePlayUrl="https://play.google.com"
      handleAllTrackEvents={() => {}}
    />
  </>
);

Layout.defaultProps = {
  avatar: null,
  children: [],
  siteName: 'Open Edx',
  siteUrl: 'https://open.edx.org/',
  username: null,
};

Layout.propTypes = {
  avatar: PropTypes.string,
  children: PropTypes.node,
  siteName: PropTypes.string,
  siteUrl: PropTypes.string,
  username: PropTypes.string,
};

const ConnectedLayout = connect(state => ({
  avatar: state.userAccount.profileImage.imageUrlMedium,
  username: state.authentication.username,
}))(Layout);

export default ConnectedLayout;
