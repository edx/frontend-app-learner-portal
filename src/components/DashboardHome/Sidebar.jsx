import React from 'react';
import { sendTrackEvent } from '@edx/frontend-analytics';
import { faExternalLinkAlt } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import Links from '../Links/Links';
import linksData from './sampleLinks';
import SidebarBlock from './SidebarBlock';

const Sidebar = () => (
  <>
    <SidebarBlock title="Program Documents" className="mb-5">
      <Links id={linksData.id} links={linksData.links} label="program documents" />
    </SidebarBlock>
    <SidebarBlock title="Manage Your Degree" className="mb-5">
      <p>Go to Georgia Tech portal to</p>
      <ul>
        <li>Add or drop courses</li>
        <li>Finance department</li>
        <li>Contact an advisor</li>
        <li>Get your grade</li>
        <li>Program wide discussions</li>
        <li>and more</li>
      </ul>
      <p>
        <a
          href="https://www.edx.org/"
          target="_blank"
          rel="noopener noreferrer"
          onClick={() => { sendTrackEvent('edx.learner_portal.school_portal_link.clicked'); }}
        >
          Go to Georgia Tech portal
          <FontAwesomeIcon
            className="ml-2 text-primary"
            icon={faExternalLinkAlt}
            size="sm"
            aria-hidden={false}
            aria-label="opens in a new window"
          />
        </a>
      </p>
    </SidebarBlock>
    <SidebarBlock title="Get Technical Support">
      <p>
        <a
          href="https://www.edx.org/"
          target="_blank"
          rel="noopener noreferrer"
          onClick={() => { sendTrackEvent('edx.learner_portal.support_link.clicked'); }}
        >
          Go to edX help center
          <FontAwesomeIcon
            className="ml-2 text-primary"
            icon={faExternalLinkAlt}
            size="sm"
            aria-hidden={false}
            aria-label="opens in a new window"
          />
        </a>
      </p>
    </SidebarBlock>
  </>
);

export default Sidebar;
