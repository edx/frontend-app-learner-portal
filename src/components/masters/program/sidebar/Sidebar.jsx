import React from 'react';
import PropTypes from 'prop-types';
import { sendTrackEvent } from '@edx/frontend-analytics';
import { faExternalLinkAlt } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import Links from './Links';
import SidebarBlock from '../../../common/Layout/SidebarBlock';

const Sidebar = ({ programDocuments, externalProgramWebsite }) => (
  <>
    {programDocuments && programDocuments.display &&
      <SidebarBlock title={programDocuments.header} className="mb-5">
        <Links
          id={programDocuments.header.toLowerCase().split(' ').join('-')}
          links={programDocuments.documents}
          label="program documents"
        />
      </SidebarBlock>
    }
    {externalProgramWebsite && externalProgramWebsite.display &&
      <SidebarBlock title={externalProgramWebsite.header} className="mb-5">
        {/* eslint-disable-next-line react/no-danger */}
        <div dangerouslySetInnerHTML={{ __html: externalProgramWebsite.description }} />
        <p>
          <a
            href={externalProgramWebsite.link.url}
            target="_blank"
            rel="noopener noreferrer"
            onClick={() => {
              sendTrackEvent('edx.learner_portal.school_portal_link.clicked');
            }}
          >
            {externalProgramWebsite.link.display_text}
            <FontAwesomeIcon
              className="ml-2 text-primary"
              icon={faExternalLinkAlt}
              size="sm"
              aria-hidden={false}
              aria-label="opens in a new window"
            />
          </a>
        </p>
      </SidebarBlock>
    }
    <SidebarBlock title="Get Technical Support">
      <p>
        <a
          href="https://support.edx.org/hc/en-us"
          target="_blank"
          rel="noopener noreferrer"
          onClick={() => {
            sendTrackEvent('edx.learner_portal.support_link.clicked');
          }}
        >
          Go to edX help center
          <FontAwesomeIcon
            className="ml-2 text-primary"
            icon={faExternalLinkAlt}
            size="sm"
            aria-hidden={false}
            aria-label="opens in a new window"
          />
        </a>
      </p>
    </SidebarBlock>
  </>
);

Sidebar.defaultProps = {
  programDocuments: null,
  externalProgramWebsite: null,
};

Sidebar.propTypes = {
  programDocuments: PropTypes.shape({
    display: PropTypes.bool,
    header: PropTypes.string,
    documents: PropTypes.arrayOf(PropTypes.shape({
      display_text: PropTypes.string,
      document: PropTypes.string,
    })),
  }),
  externalProgramWebsite: PropTypes.shape({
    header: PropTypes.string,
    link: PropTypes.shape({
      display_text: PropTypes.string,
      url: PropTypes.string,
    }),
    display: PropTypes.bool,
    description: PropTypes.string,
  }),
};

export default Sidebar;
