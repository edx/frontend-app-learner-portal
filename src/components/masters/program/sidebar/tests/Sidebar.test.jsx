import React from 'react';
import renderer from 'react-test-renderer';

import { LayoutContext } from '../../../../common/layout';

import Sidebar from '../Sidebar';

describe('<Sidebar />', () => {
  it('renders correctly', () => {
    const pageContext = {
      programDocuments: null,
      externalProgramWebsite: 'https://edx.org',
    };
    const tree = renderer
      .create((
        <LayoutContext.Provider value={{ pageContext }}>
          <Sidebar />
        </LayoutContext.Provider>
      ))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with links', () => {
    const pageContext = {
      programDocuments: {
        header: 'Program Documents',
        display: true,
        documents: [
          {
            display_text: 'Test Document',
            document: '/test/document',
          },
          {
            display_text: 'Test Link',
            url: 'example.com/1',
          },
          {
            display_text: 'Test Document 2',
            document: '/test/document2',
          },
        ],
      },
      externalProgramWebsite: 'https://edx.org',
    };
    const tree = renderer.create((
      <LayoutContext.Provider value={{ pageContext }}>
        <Sidebar />
      </LayoutContext.Provider>
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
