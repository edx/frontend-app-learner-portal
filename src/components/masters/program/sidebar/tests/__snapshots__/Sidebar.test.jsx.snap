// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> renders correctly 1`] = `
Array [
  <div
    className="mb-5"
  >
    <h2
      className="mb-2"
    >
      Program Documents
    </h2>
    <nav
      aria-label="program documents"
    >
      <ul
        className="list-unstyled mb-2"
        id="program-documents"
      >
        <li
          className="mb-1"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-file fa-w-12 mr-2 text-primary"
            data-icon="file"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224 136V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V160H248c-13.2 0-24-10.8-24-24zm160-14.1v6.1H256V0h6.1c6.4 0 12.5 2.5 17 7l97.9 98c4.5 4.5 7 10.6 7 16.9z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <a
            href="https://edx.org/degree-requirements"
            onClick={[Function]}
            rel="noopener noreferrer"
            target="_blank"
          >
            Degree Requirements
          </a>
        </li>
        <li
          className="mb-1"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-file fa-w-12 mr-2 text-primary"
            data-icon="file"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224 136V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V160H248c-13.2 0-24-10.8-24-24zm160-14.1v6.1H256V0h6.1c6.4 0 12.5 2.5 17 7l97.9 98c4.5 4.5 7 10.6 7 16.9z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <a
            href="https://edx.org/student-record"
            onClick={[Function]}
            rel="noopener noreferrer"
            target="_blank"
          >
            Student Record
          </a>
        </li>
        <li
          className="mb-1"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-file fa-w-12 mr-2 text-primary"
            data-icon="file"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224 136V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V160H248c-13.2 0-24-10.8-24-24zm160-14.1v6.1H256V0h6.1c6.4 0 12.5 2.5 17 7l97.9 98c4.5 4.5 7 10.6 7 16.9z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <a
            href="https://edx.org/test-1"
            onClick={[Function]}
            rel="noopener noreferrer"
            target="_blank"
          >
            Test 1
          </a>
        </li>
        <li
          className="mb-1"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-file fa-w-12 mr-2 text-primary"
            data-icon="file"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224 136V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V160H248c-13.2 0-24-10.8-24-24zm160-14.1v6.1H256V0h6.1c6.4 0 12.5 2.5 17 7l97.9 98c4.5 4.5 7 10.6 7 16.9z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <a
            href="https://edx.org/test-2"
            onClick={[Function]}
            rel="noopener noreferrer"
            target="_blank"
          >
            Test 2
          </a>
        </li>
        <li
          className="mb-1"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-file fa-w-12 mr-2 text-primary"
            data-icon="file"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224 136V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V160H248c-13.2 0-24-10.8-24-24zm160-14.1v6.1H256V0h6.1c6.4 0 12.5 2.5 17 7l97.9 98c4.5 4.5 7 10.6 7 16.9z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <a
            href="https://edx.org/test-3"
            onClick={[Function]}
            rel="noopener noreferrer"
            target="_blank"
          >
            Test 3
          </a>
        </li>
      </ul>
    </nav>
    <button
      aria-controls="program-documents"
      aria-expanded={false}
      className="btn toggle-show-all-btn px-0 btn-link"
      onBlur={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-chevron-circle-down fa-w-16 mr-2"
        data-icon="chevron-circle-down"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M504 256c0 137-111 248-248 248S8 393 8 256 119 8 256 8s248 111 248 248zM273 369.9l135.5-135.5c9.4-9.4 9.4-24.6 0-33.9l-17-17c-9.4-9.4-24.6-9.4-33.9 0L256 285.1 154.4 183.5c-9.4-9.4-24.6-9.4-33.9 0l-17 17c-9.4 9.4-9.4 24.6 0 33.9L239 369.9c9.4 9.4 24.6 9.4 34 0z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <span>
        show all 8
      </span>
    </button>
  </div>,
  <div
    className="mb-5"
  >
    <h2
      className="mb-2"
    >
      Manage Your Degree
    </h2>
    <p>
      Go to Georgia Tech portal to
    </p>
    <ul>
      <li>
        Add or drop courses
      </li>
      <li>
        Finance department
      </li>
      <li>
        Contact an advisor
      </li>
      <li>
        Get your grade
      </li>
      <li>
        Program wide discussions
      </li>
      <li>
        and more
      </li>
    </ul>
    <p>
      <a
        href="https://www.edx.org/"
        onClick={[Function]}
        rel="noopener noreferrer"
        target="_blank"
      >
        Go to Georgia Tech portal
        <svg
          aria-hidden={false}
          aria-label="opens in a new window"
          className="svg-inline--fa fa-external-link-alt fa-w-18 fa-sm ml-2 text-primary"
          data-icon="external-link-alt"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M576 24v127.984c0 21.461-25.96 31.98-40.971 16.971l-35.707-35.709-243.523 243.523c-9.373 9.373-24.568 9.373-33.941 0l-22.627-22.627c-9.373-9.373-9.373-24.569 0-33.941L442.756 76.676l-35.703-35.705C391.982 25.9 402.656 0 424.024 0H552c13.255 0 24 10.745 24 24zM407.029 270.794l-16 16A23.999 23.999 0 0 0 384 303.765V448H64V128h264a24.003 24.003 0 0 0 16.97-7.029l16-16C376.089 89.851 365.381 64 344 64H48C21.49 64 0 85.49 0 112v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V287.764c0-21.382-25.852-32.09-40.971-16.97z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </a>
    </p>
  </div>,
  <div>
    <h2
      className="mb-2"
    >
      Get Technical Support
    </h2>
    <p>
      <a
        href="https://www.edx.org/"
        onClick={[Function]}
        rel="noopener noreferrer"
        target="_blank"
      >
        Go to edX help center
        <svg
          aria-hidden={false}
          aria-label="opens in a new window"
          className="svg-inline--fa fa-external-link-alt fa-w-18 fa-sm ml-2 text-primary"
          data-icon="external-link-alt"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M576 24v127.984c0 21.461-25.96 31.98-40.971 16.971l-35.707-35.709-243.523 243.523c-9.373 9.373-24.568 9.373-33.941 0l-22.627-22.627c-9.373-9.373-9.373-24.569 0-33.941L442.756 76.676l-35.703-35.705C391.982 25.9 402.656 0 424.024 0H552c13.255 0 24 10.745 24 24zM407.029 270.794l-16 16A23.999 23.999 0 0 0 384 303.765V448H64V128h264a24.003 24.003 0 0 0 16.97-7.029l16-16C376.089 89.851 365.381 64 344 64H48C21.49 64 0 85.49 0 112v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V287.764c0-21.382-25.852-32.09-40.971-16.97z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </a>
    </p>
  </div>,
]
`;
